pkgs/misc/vscode-extensions/default.nix

```diff
@@ -245,11 +245,19 @@ let
       llvm-org.lldb-vscode = llvmPackages_8.lldb;
 
       WakaTime.vscode-wakatime = callPackage ./wakatime {};
-    } // lib.optionalAttrs (config.allowAliases or true) (
-    with self; {
+    };
+
+    aliases = self: super: {
       # aliases
-      ms-vscode.Go = golang.Go;
-    }
-  );
+      ms-vscode = lib.recursiveUpdate super.ms-vscode { inherit (super.golang) Go; };
```

Comment:
For me this approach feels a little fragile:
- If all ms-vscode extensions are dropped, the aliases need to be changed, because super.ms-vscode does not exist anymore. Right?
- If then a ms-vscode extension is added again, it is not there until the aliases overlay is adjusted again.
Is it possible to use mkMerge or something like this here?

Reply:
Ah, you did exactly that in the other PR.

Reply:
> * If all ms-vscode extensions are dropped, the aliases need to be changed, because super.ms-vscode does not exist anymore. Right?
> 
> * If then a ms-vscode extension is added again, it is not there until the aliases overlay is adjusted again.

This shouldn't be too much of an issue, because this is all co-located within the same file. The more elegant implementation is in the other PR, but it might be "controversial" to add a recursiveExtends function.